alembic/versions/5fd7e8758fb1_add_autoincrement_to_device_tables.py

```diff
@@ -0,0 +1,143 @@
+"""add_autoincrement_to_device_tables
+
+Revision ID: 5fd7e8758fb1
+Revises: fe23de4ed528
+Create Date: 2024-05-21 09:25:47.751986
+
+"""
+
+import sqlalchemy as sa
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "5fd7e8758fb1"
+down_revision = "fe23de4ed528"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # Modify illumina flow cell primary key
+    op.drop_constraint(
+        constraint_name="illumina_flow_cell_ibfk_1",
+        table_name="illumina_flow_cell",
+        type_="foreignkey",
+    )
+    op.drop_constraint(constraint_name="fk_device_id", table_name="run_metrics", type_="foreignkey")
+    op.execute("ALTER TABLE run_device MODIFY COLUMN id INT NOT NULL AUTO_INCREMENT")
+    op.alter_column(
```

Comment:
the model column was set to not nullable originally, this changes it to nullable